common/test/acceptance/tests/lms/test_bookmarks.py

```diff
@@ -0,0 +1,196 @@
+    ]
+
+    def create_course_fixture(self):
+        """ Create course fixture """
+        self.course_fixture = CourseFixture(  # pylint: disable=attribute-defined-outside-init
+            self.course_info['org'], self.course_info['number'],
+            self.course_info['run'], self.course_info['display_name']
+        )
+
+        self.course_fixture.add_children(
+            XBlockFixtureDesc('chapter', self.PROBLEMS_INFO[0][0]).add_children(
+                XBlockFixtureDesc('sequential', self.PROBLEMS_INFO[0][1]).add_children(
+                    XBlockFixtureDesc('problem', self.PROBLEMS_INFO[0][2])
+                )
+            ),
+            XBlockFixtureDesc('chapter', self.PROBLEMS_INFO[1][0]).add_children(
+                XBlockFixtureDesc('sequential', self.PROBLEMS_INFO[1][1]).add_children(
+                    XBlockFixtureDesc('problem', self.PROBLEMS_INFO[1][2])
+                )
+            )
+        ).install()
+
+
+class BookmarksTest(BookmarksTestMixin):
+    """
+    Tests to verify bookmarks functionality.
+    """
+
+    def setUp(self):
+        """
+        Initialize test setup.
+        """
+        super(BookmarksTest, self).setUp()
+
+        self.course_outline_page = CourseOutlinePage(
+            self.browser,
+            self.course_info['org'],
+            self.course_info['number'],
+            self.course_info['run']
+        )
+
+        self.create_course_fixture()
+
+        # Auto-auth register for the course.
+        AutoAuthPage(self.browser, username=self.USERNAME, email=self.EMAIL, course_id=self.course_id).visit()
+
+        self.courseware_page = CoursewarePage(self.browser, self.course_id)
+        self.courseware_page.visit()
+        self.bookmarks = BookmarksPage(self.browser, self.course_id)
+
+        # Use auto-auth to retrieve the session for a logged in user
+        self.session = requests.Session()
+        response = self.session.get(LMS_BASE_URL + "/auto_auth?username=STUDENT&email=student@example.com")
+        self.assertTrue(response.ok, "Failed to get session info")
+
+    def _bookmark_unit(self, course_id, usage_id):
+        """ Bookmark a single unit """
+        csrftoken = self.session.cookies['csrftoken']
+        headers = {'Content-type': 'application/json', "X-CSRFToken": csrftoken}
+        url = LMS_BASE_URL + "/api/bookmarks/v0/bookmarks/?course_id=" + course_id + '&fields=path'  # pylint: disable=protected-access
+        data = json.dumps({'usage_id': usage_id})
+
+        response = self.session.post(url, data=data, headers=headers, cookies=self.session.cookies)
+        response = json.loads(response.text)
+        self.assertTrue(response['usage_id'] == usage_id, "Failed to bookmark unit")
+
+    def _bookmark_all_units(self, xblocks):
+        """ Bookmark all units in a course """
+        for xblock in xblocks:
+            self._bookmark_unit(self.course_id, usage_id=xblock.locator)
+
+    def _delete_section(self, index=0):
```

Comment:
It would be better to not have a default value for index; it should be specified explicitly by the caller.